### ISSUE TITLE
[Perf][Kernel] stride-aware kernels to drop contiguous() copy on dim != -1

### ISSUE THREAD

**Issue by lcy-seso** [tile-ai/TileOPs#1065]
### Symptom / Motivation

Canonical-pattern ops migrating off the legacy `(M, N)` ctor (PRs #1061, #1064 and future scan / norm / reduction / softmax migrations) include the pipeline:

```python
if dim_norm != ndim - 1:
    x = x.movedim(dim_norm, -1)
post_move_shape = tuple(x.shape)
x = x.contiguous().reshape(-1, self.N)   # full read+write copy when dim != -1 or input non-contiguous
```

`x.contiguous()` triggers a full tensor copy whenever the input is non-contiguous along the inner dim — i.e., whenever the user passes `dim != -1` (default) or feeds a sliced view. For typical LLM workloads `dim = -1` is universal and the cost is zero, but it is still a known opt-in tax baked into every canonical-pattern op.

Surfaced by a gemini-code-assist review on PR #1064.

### Root Cause Analysis

Tilelang kernels (`RMSNormKernel`, `CumulativeKernel`, …) currently assume the input is row-major contiguous `(M, N)` 2-D. When the manifest declares `dim` as a `signature.params` entry (canonical pattern per `docs/ops-design.md` § Step 3), the op layer is the only place that can normalize layout — hence the host-side `movedim + contiguous + reshape`.

The principled fix is at the kernel ABI: pass strides through and let the kernel issue strided loads, removing the host-side copy.

### Related Files

- `tileops/ops/norm/norm_base.py` — `RowNormOp._validate_and_normalize_dim` / `_flatten_to_2d`
- `tileops/ops/reduction/cumulative_base.py` — `CumulativeOp._run`
- `tileops/kernels/norm/rms_norm.py` — first prototype target (smallest kernel, single dispatch, well-tested)
- `tileops/kernels/reduction/cumulative.py` — second target after RMS proves the pattern

## Goal

Selected row-norm / scan / reduction kernels accept `(x, M, N, stride_M, stride_N)` and use `T.copy` with explicit strides instead of assuming row-major contiguous layout. Op-layer canonical-pattern bases drop the `x.contiguous()` call when feeding these kernels and pass the strides through verbatim. Net effect: zero host-side copy on `dim != -1` workloads, no regression on `dim = -1`.

## Plan

**Plan type: proposal**

1. Prototype on `RMSNormKernel`: extend ctor / build to accept `stride_M, stride_N` (default `(N, 1)` for backwards-compat), thread through to `T.copy`. Keep existing call sites unchanged.
2. Add a stride-aware path in `RowNormOp._run`: when `dim != -1`, skip `contiguous()`, pass `x.stride(dim_norm), x.stride(...)` to the kernel.
3. Measure perf on a representative `dim != -1` workload (e.g., RMS over a non-last axis on a Llama-style 3-D shape). Compare against the contiguous-copy baseline.
4. Decide GO / NO-GO based on the measurement: GO → roll out the same pattern to `CumulativeKernel` (`CumulativeOp._run`) and any future canonical-pattern kernels; NO-GO → revert the kernel change, document the cost as accepted.
5. If GO: file a follow-up issue listing every canonical-pattern kernel that needs the stride-aware ABI.

## Constraints

- **Inner-dim contiguity (`stride_N == 1`) is non-negotiable** for coalesced GPU loads. `movedim(dim, -1)` only changes which axis is inner, not the inner axis's contiguity, so this is fine — but the kernel ABI must enforce it (assert / static guard).
- **Kernel-layer only**. This issue tracks the kernel ABI change. The op-layer canonical pattern (PRs #1061, #1064) ships as-is — no rollback there.
- **No regression on `dim = -1`**. The default path must be free (`stride_M = N, stride_N = 1` is the contiguous case; tile-level codegen should compile to the same loads).

## Acceptance Criteria

- [ ] AC-1: Modified files pass existing unit tests.
- [ ] AC-2: `RMSNormKernel` accepts optional `stride_M, stride_N` ctor args; existing call sites with no stride args produce identical output (bit-exact on `dim = -1` workloads).
- [ ] AC-3: A new test in `tests/ops/test_rms_norm.py` exercises a `dim != -1` shape (e.g., 3-D `(B, H, S)` with `dim = 1`) using the stride-aware path and matches the PyTorch reference within the existing `atol`.
- [ ] AC-4: Measured perf delta on a representative `dim != -1` workload is reported in the PR (e.g., `(2, 4096, 512)` bf16 with `dim = 1` on H200), comparing the new stride-aware path vs. the host-side `contiguous()` baseline.
- [ ] AC-5: `RowNormOp._run` either drops `x.contiguous()` (GO path) or keeps it with a documented reason (NO-GO path).
- [ ] AC-6: Decision (GO / NO-GO) and follow-up scope (which kernels to extend) recorded in the PR description.
